File: csharp/Deck.cs
using System;
using System.Collections.Generic;

public class Deck {
    	private List<Card> cards; 

    	// Deck constructor 
	public Deck() {
        	cards = new List<Card>();
		for (int rank = 2; rank <= 14; rank++) {
			cards.Add(new Card(rank, 'D'));
			cards.Add(new Card(rank, 'C'));
			cards.Add(new Card(rank, 'H'));
			cards.Add(new Card(rank, 'S'));
		}
    	}

	// randomly shuffles deck of cards 
    	public void Shuffle() {
		// creates a new instance of the Random class to generate random numbers
        	Random rng = new Random(); 
        	int n = cards.Count;
        	while (n > 1) {
            		int k = rng.Next(n--); // generates a random index 
            		Card value = cards[k];
            		cards[k] = cards[n];
            		cards[n] = value;
        	}
    	}

	// method to deal 6 hands of 5 cards 
    	public void DealHands(List<List<Card>> hands) {
        	for (int i = 0; i < 5; i++) {
			for (int j = 0; j < hands.Count; j++) {
				hands[j].Add(cards[0]);
				cards.RemoveAt(0);
			}
		}
    	}
	
	// method to print the cards in the deck, singleLine bool used to start a new line
    	public void PrintDeck(bool singleLine) {
        	int count = 0;
		foreach (var card in cards) {
			Console.Write(card.ToString() + " ");
			count++;
			if (!singleLine && count % 13 == 0) {
				Console.WriteLine();
			}
		}
		Console.WriteLine();
    	}
}

File: go/Deck.go
package main

import (
	"fmt"
    	"math/rand"
    	"time"
)

// Deck struct
type Deck struct {
    	Cards []Card
}

// Creates and returns a new deck of 52 cards
func NewDeck() Deck {
    	suits := []Suit{Hearts, Diamonds, Clubs, Spades}
	ranks := []string{"2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K", "A"}
    	var cards []Card

    	for _, suit := range suits {
        	for _, rank := range ranks {
            		card, err := NewCard(rank, suit)
            		if err == nil {
                		cards = append(cards, card)
            		}
        	}
    	}

    	return Deck{Cards: cards}
}

// Shuffles the deck 
func (d *Deck) Shuffle() {
	rand.Seed(time.Now().UnixNano())
    	rand.Shuffle(len(d.Cards), func(i, j int) {
        	d.Cards[i], d.Cards[j] = d.Cards[j], d.Cards[i]
    	})
}

// Deals 6 hands of 5 cards each from the deck
func (d *Deck) DealHands(hands [][]Card) {
    	for i := 0; i < 5; i++ {
        	for j := 0; j < len(hands); j++ {
            		hands[j] = append(hands[j], d.Cards[0])
            		d.Cards = d.Cards[1:]  // Remove the dealt card from the deck
        	}
    	}
}

// Prints the cards in the deck
func (d *Deck) PrintDeck(singleLine bool) {
    	count := 0
    	for _, card := range d.Cards {
        	fmt.Print(card.ToString() + " ")
        	count++
        	if !singleLine && count%13 == 0 {
        		fmt.Println()
        	}
    	}
    	fmt.Println()
}

